File: app/src/main/java/com/example/healthtracker/model/CalcDao.java
package com.example.healthtracker.model;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface CalcDao {

    @Insert
    public void insert(Calc calc);

    @Query("SELECT * from Calc where type = :type")
    public List<Calc> getRegisterByType(String type);

    @Query("DELETE from Calc where type = :type")
    public void deleteAllByType(String type);

}

File: app/src/main/java/com/example/healthtracker/model/CalcDao.kt
package com.example.healthtracker.model

import androidx.room.*

@Dao
interface CalcDao {

    @Insert
    fun insert(calc: Calc)

    @Query("SELECT * from Calc where type= :type")
    fun getRegisterByType(type: String): List<Calc>

    @Query("DELETE from Calc where type = :type")
    fun deleteAllByType(type: String)

}